components/suggest/src/store.rs

```diff
@@ -337,16 +340,45 @@ where
                     .get_records_with_options(&options)?
                     .records;
 
-                self.ingest_records(writer, &records_chunk)?;
+                self.ingest_records(LAST_INGEST_META_UNPARSABLE, writer, &records_chunk)?;
             }
         }
 
+        let mut ingest_record_types = if let Some(rt) = &constraints.providers {
+            rt.iter().flat_map(|x| x.records_for_provider()).collect()
+        } else {
+            DEFAULT_RECORDS_TYPES.to_vec()
+        };
+        ingest_record_types.sort_unstable();
+        ingest_record_types.dedup();
```

Comment:
This does the deduping and maintains the order. Also no need to dedup the constant type array. 
```suggestion
        // use std::collections::BTreeSet;
        let ingest_record_types = if let Some(rt) = &constraints.providers {
            rt.iter()
                .flat_map(|x| x.records_for_provider())
                .collect::<BTreeSet<_>>()
                .into_iter()
                .collect()
        } else {
            DEFAULT_RECORDS_TYPES.to_vec()
        };
```